modules/stochastic_tools/include/state_sim/model/actions/StateSimAction.h

```diff
@@ -0,0 +1,38 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+
+#ifndef STATESIMACTION_H
+#define STATESIMACTION_H
+
+#include "StateSimBase.h"
+#include <string>
+
+class StateSimAction : public StateSimBase
+{
+protected:
+  ACTION_TYPE_ENUM _act_type;
+
+private:
+  static unsigned int _counter;
```

Comment:
There is actually still a problem with statics and Multiapps. If you have an app and a Multiapp both with a state model active, you'll still only have the one global counter. It'll reflect the total count of all objects in both the main and multiapp instances instead of giving you accurate counts for each. This will raise havoc on your bitset indexing.